www_src/lib/spec.js

```diff
@@ -0,0 +1,147 @@
+var React = require('react/addons');
+
+function getValidationErrors(spec) {
+  var result = null;
+  var VALID_KEYS = ['category', 'default', 'validation'];
+  var VALID_CATEGORIES = ['styles', 'attributes'];
+
+  var keys = Object.keys(spec);
+
+  if (!keys.length) {
+    result = 'Spec must have at least one prop';
+  }
+
+  keys.forEach(key => {
+    var prop = spec[key];
+    Object.keys(prop).forEach(propKey => {
+      if (VALID_KEYS.indexOf(propKey) <= -1) result = key + ' is not a valid key';
+    });
+    if (VALID_CATEGORIES.indexOf(prop.category) <= -1) {
+      result = prop.category + ' is not a valid category definition for ' + key;
+    }
+  });
+
+  return result;
+}
+
+function Spec(type, spec) {
+  if (typeof type !== 'string' || typeof spec !== 'object') throw new Error('Spec takes two params: type (String) and spec (Object)');
+  var error = getValidationErrors(spec);
+  if (error) throw new Error(error);
```

Comment:
Note that if a spec is defined in a block and it doesn't pass validations (e.g. uses a category other than `styles` or `attributes`, see tests for more detail), it will throw